include/aws/s3/s3_client.h

```diff
@@ -1390,6 +1390,17 @@ int aws_s3_request_metrics_get_request_id(
     const struct aws_s3_request_metrics *metrics,
     const struct aws_string **out_request_id);
 
+/**
+ * Get the extended request ID from aws_s3_request_metrics.
+ * If unavailable, AWS_ERROR_S3_METRIC_DATA_NOT_AVAILABLE will be raised.
+ * If available, out_amz_id_2 will be set to a string. Be warned this string's lifetime is tied to the metrics
+ * object.
+ **/
+AWS_S3_API
+int aws_s3_request_metrics_get_amz_id_2(
```

Comment:
let's rename this to extended request id. 
https://docs.aws.amazon.com/AmazonS3/latest/API/get-request-ids.html#sdk-request-ids
Looks like that is used in the s3 public docs.

The amz-id-2 is the header name.